lib/agent.js

```diff
@@ -144,37 +144,41 @@ Agent.prototype.start = function (opts) {
     })
   }
 
-  this._apmServer = new ElasticAPMHttpClient({
-    // metadata
-    agentName: 'nodejs',
-    agentVersion: version,
-    serviceName: this._conf.serviceName,
-    serviceVersion: this._conf.serviceVersion,
-    frameworkName: this._conf.frameworkName,
-    frameworkVersion: this._conf.frameworkVersion,
-    hostname: this._conf.hostname,
-
-    // Sanitize conf
-    truncateKeywordsAt: config.INTAKE_STRING_MAX_SIZE,
-    truncateErrorMessagesAt: this._conf.errorMessageMaxLength,
-
-    // HTTP conf
-    secretToken: this._conf.secretToken,
-    userAgent: userAgent,
-    serverUrl: this._conf.serverUrl,
-    rejectUnauthorized: this._conf.verifyServerCert,
-    serverTimeout: this._conf.serverTimeout * 1000,
-
-    // Streaming conf
-    size: this._conf.apiRequestSize,
-    time: this._conf.apiRequestTime * 1000,
-
-    // Debugging
-    payloadLogFile: this._conf.payloadLogFile
-  })
-  this._apmServer.on('error', err => {
-    this.logger.error('An error occrued while communicating with the APM Server:', err.message)
-  })
+  if (this._conf.transport) {
+    this._apmServer = this._conf.transport(this._conf)
+  } else {
+    this._apmServer = new ElasticAPMHttpClient({
+      // metadata
+      agentName: 'nodejs',
+      agentVersion: version,
+      serviceName: this._conf.serviceName,
+      serviceVersion: this._conf.serviceVersion,
+      frameworkName: this._conf.frameworkName,
+      frameworkVersion: this._conf.frameworkVersion,
+      hostname: this._conf.hostname,
+
+      // Sanitize conf
```

Comment:
It would be nice if we could put the sanitation logic in a generic `Transport` object that all transports inherited from

Reply:
Do we want this PR to depend on that, or should build and land that independently? 🤔

Reply:
As long as we don't document the new `transport` config option, then I'm ok with landing it independently